core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -187,7 +187,12 @@ bool detray::rk_stepper<magnetic_field_t, transform3_t, constraint_t, policy_t,
     scalar error_estimate = 0;
 
     // First Runge-Kutta point
-    sd.b_first = _magnetic_field.get_field(stepping().pos(), context_type{});
+    const vector3 spos = stepping().pos();
+    const typename magnetic_field_t::output_t bvec = _magnetic_field.at(spos[0], spos[1], spos[2]);
```

Comment:
In this case, that will be an `std::array`, but this behaviour is configurable: can be a detray vector too (but let's do that later, as it will also require some very slight modifications in covfie itself).